internal/pkg/addon/package.go

```diff
@@ -0,0 +1,276 @@
+// other than the AWS::Serverless resources, which are not supported in Copilot.
+//
+// TODO(dnrnd) AWS::Include.Location
+var resourcePackageConfig = map[string][]packagePropertyConfig{
+	"AWS::ApiGateway::RestApi": {
+		{
+			Property:           []string{"BodyS3Location"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+		},
+	},
+	"AWS::Lambda::Function": {
+		{
+			Property:           []string{"Code"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+			ForceZip:           true,
+		},
+	},
+	"AWS::Lambda::LayerVersion": {
+		{
+			Property:           []string{"Content"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+			ForceZip:           true,
+		},
+	},
+	"AWS::Serverless::Function": {
+		{
+			Property: []string{"CodeUri"},
+			ForceZip: true,
+		},
+	},
+	"AWS::Serverless::LayerVersion": {
+		{
+			Property: []string{"ContentUri"},
+			ForceZip: true,
+		},
+	},
+	"AWS::AppSync::GraphQLSchema": {
+		{
+			Property: []string{"DefinitionS3Location"},
+		},
+	},
+	"AWS::AppSync::Resolver": {
+		{
+			Property: []string{"RequestMappingTemplateS3Location"},
+		},
+		{
+			Property: []string{"ResponseMappingTemplateS3Location"},
+		},
+	},
+	"AWS::AppSync::FunctionConfiguration": {
+		{
+			Property: []string{"RequestMappingTemplateS3Location"},
+		},
+		{
+			Property: []string{"ResponseMappingTemplateS3Location"},
+		},
+	},
+	"AWS::Serverless::Api": {
+		{
+			Property: []string{"DefinitionUri"},
+		},
+	},
+	"AWS::ElasticBeanstalk::ApplicationVersion": {
+		{
+			Property:           []string{"SourceBundle"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+		},
+	},
+	"AWS::CloudFormation::Stack": {
+		{
+			// This implementation does not recursively package
+			// the local template pointed to by TemplateURL.
+			Property: []string{"TemplateURL"},
+		},
+	},
+	"AWS::Glue::Job": {
+		{
+			Property: []string{"Command", "ScriptLocation"},
+		},
+	},
+	"AWS::StepFunctions::StateMachine": {
+		{
+			Property:           []string{"DefinitionS3Location"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+		},
+	},
+	"AWS::Serverless::StateMachine": {
+		{
+			Property:           []string{"DefinitionUri"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+			ForceZip:           true,
+		},
+	},
+	"AWS::CodeCommit::Repository": {
+		{
+			Property:           []string{"Code", "S3"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+			ForceZip:           true,
+		},
+	},
+}
+
+func (t *cfnTemplate) packageTemplate(a *Addons) error {
+	resources := mappingNode(&t.Resources)
+
+	for name, node := range resources {
+		resType := yamlMapGet(node, "Type").Value
+		confs, ok := resourcePackageConfig[resType]
+		if !ok {
+			continue
+		}
+
+		props := yamlMapGet(node, "Properties")
+		for _, conf := range confs {
+			if err := a.packageProperty(props, conf); err != nil {
```

Comment:
this ping-pong between `addons -> cfnTemplate.package -> addons.packageProperty` is kinda funny 🤔 I can't really think of a better way so :shipit:

Reply:
yeah...i didn't like adding 2 arguments to every function `(bucket, uploader)`, so i considered making `Addons` meet this 
```
type bucketUploader interface {
  upload(key, data)
}
```
and that way i'd only pass one which wasn't as bad but also a lot of extra stuff for.....not much benefit/readability imo 😊